scout/parse/variant/clnsig.py

```diff
@@ -3,9 +3,47 @@
 
 import cyvcf2
 
+from scout.constants.clnsig import ONC_CLNSIG
+
 LOG = logging.getLogger(__name__)
 
 
+def split_groups(value: str) -> List[str]:
+    """Removes leading underscore from a string and splits it into a list of items."""
+    return [
+        item.lstrip("_").replace(" ", "_")
+        for group in value.replace("&", ",").split(",")
+        for item in group.split("/")
+    ]
+
+
+def parse_clnsig_onc(variant: cyvcf2.Variant) -> List[dict]:
+    """Collect somatic oncogenicity ClinVar classifications for a variant, if available."""
+    if not variant.INFO.get("ONC"):
+        return []
+    acc = int(variant.INFO.get("CLNVID", 0))
+    onc_sig_groups = split_groups(value=variant.INFO.get("ONC", "").lower())
+    onc_revstat = ",".join(split_groups(value=variant.INFO.get("ONCREVSTAT", "").lower()))
+    onc_dn_groups = split_groups(variant.INFO.get("ONCDN", ""))
+
+    onc_clnsig_accessions = []
+
+    for i, onc_sig in enumerate(onc_sig_groups):
+        if (
+            onc_sig.capitalize() not in ONC_CLNSIG
+        ):  # This is excluding entries with ONC=no_classification_for_the_single_variant
+            continue
+        onc_clnsig_accessions.append(
+            {
+                "accession": acc,
+                "value": onc_sig,
+                "revstat": onc_revstat,
+                "dn": onc_dn_groups[i].replace("|", ",").replace("_", " "),
```

Comment:
Or do we want to save them like: "Neoplasm,Thyroid_gland_undifferentiated_(anaplastic)_carcinoma", to be consistent with revstat, which has underscores?

Reply:
I don't think it matters terribly as long as we are consistent ourselves: we would search using the same separator if it ever comes to that, which it might if this grows well. I have a slight feeling these categories are even less static than the revstats - the need to qualify with a parenthesis in the term kind of hints at non permanence. 😅